src/azure-cli/azure/cli/command_modules/acs/decorator.py

```diff
@@ -4702,16 +4868,28 @@ def set_up_windows_profile(self, mc: ManagedCluster) -> ManagedCluster:
             windows_admin_password,
         ) = self.context.get_windows_admin_username_and_password()
         if windows_admin_username or windows_admin_password:
+            # license
             windows_license_type = None
             if self.context.get_enable_ahub():
                 windows_license_type = "Windows_Server"
 
+            # gmsa
+            gmsa_profile = None
```

Comment:
User is allowed to set the configuration of gmsa only after windows admin user name/password is provided.

If neither user name nor password is provided, `--enable-gmsa/--gmsa-dns-server/--gmsa-root-domain-name` would be ignored (same behavior as in aks-preview).

Reply:
I think that we should raise an error instead of ignoring it. how do you think?

Reply:
OK, updated in commit [#dc72c59](https://github.com/Azure/azure-cli/pull/21472/commits/dc72c5971946cd0d9d3ac78bc846a0c14cdda14a).